src/utils/string.rs

```diff
@@ -186,3 +202,157 @@ where
+            ext: *b"TXT"
+        })
+    );
+    let mut gen = gen.skip(7);
+    assert_eq!(
+        gen.next(),
+        Some(Sfn {
+            name: *b"HELLO~10",
+            ext: *b"TXT"
+        })
+    );
+}
+
+#[test]
+fn test_sfn_generator_short() {
+    let mut gen = SfnGenerator::new("run.jpeg", Codepage::default());
+
+    assert_eq!(
+        gen.next(),
+        Some(Sfn {
+            name: *b"RUN~1   ",
+            ext: *b"JPE"
+        })
+    );
+    assert_eq!(
+        gen.next(),
+        Some(Sfn {
+            name: *b"RUN~2   ",
+            ext: *b"JPE"
+        })
+    );
+    let mut gen = gen.skip(7);
+    assert_eq!(
+        gen.next(),
+        Some(Sfn {
+            name: *b"RUN~10  ",
+            ext: *b"JPE"
+        })
+    );
+}
+
+#[cfg(any(feature = "codepage", feature = "cp437"))]
+#[test]
+fn test_sfn_generator_cp_chars_cp437() {
+    let mut gen = SfnGenerator::new("tëst.txt", Codepage::CP437);
+
+    assert_eq!(
+        gen.next(),
+        Some(Sfn {
+            name: *b"T\x89ST~1  ",
+            ext: *b"TXT"
+        })
+    );
+}
+
+#[cfg(not(any(
+    feature = "codepage",
+    feature = "cp437",
+    feature = "cp720",
+    feature = "cp737",
+    feature = "cp775",
+    feature = "cp850",
+    feature = "cp852",
+    feature = "cp855",
+    feature = "cp857",
+    feature = "cp858",
+    feature = "cp860",
+    feature = "cp861",
+    feature = "cp862",
+    feature = "cp863",
+    feature = "cp864",
+    feature = "cp865",
+    feature = "cp866",
+    feature = "cp869",
+    feature = "cp874"
+)))]
+#[test]
+fn test_sfn_generator_cp_chars_ascii() {
+    let mut gen = SfnGenerator::new("tëst.txt", Codepage);
+
+    assert_eq!(
+        gen.next(),
+        Some(Sfn {
+            name: *b"TST~1   ",
+            ext: *b"TXT"
+        })
+    );
+}
+
+#[test]
+fn test_sfn_generator_unknown_chars() {
+    let mut gen = SfnGenerator::new("😇.😈", Codepage::default());
+
+    assert_eq!(
+        gen.next(),
+        Some(Sfn {
+            name: *b"~1      ",
+            ext: *b"   "
+        })
+    );
+}
+
+#[cfg(all(test, feature = "std"))]
+fn run_gen_sfn(string: &str) -> Option<Sfn> {
+    use crate::FSOptions;
+    use embedded_io_adapters::std::FromStd;
+    use std::io::Cursor;
+
+    const FAT16: &[u8] = include_bytes!("../../imgs/fat16.img");
+    let mut storage = FromStd::new(Cursor::new(FAT16.to_owned()));
+    let fs = FileSystem::new(&mut storage, FSOptions::new()).unwrap();
+
+    gen_sfn(string, &fs, "/").ok()
+}
+
+#[cfg(all(test, feature = "std"))]
+#[test]
+fn test_gen_sfn_match() {
+    assert_eq!(
+        run_gen_sfn("TEST.TXT"),
+        Some(Sfn {
+            name: *b"TEST    ",
+            ext: *b"TXT"
+        })
+    )
+}
+
+#[cfg(all(test, feature = "std"))]
+#[test]
+fn test_gen_sfn_mismatch() {
+    assert_eq!(
+        run_gen_sfn("test.txt"),
+        Some(Sfn {
+            name: *b"TEST~1  ",
+            ext: *b"TXT"
+        })
+    )
+}
```

Comment:
You mentioned before that those tests failed, yet now they seem to pass, at least on my PC. Did you fix them?

Reply:
if you check out the commit where it's introduced you'll see failures.

Reply:
Yeah, but I run those tests you introduced locally and they all seem to pass.
<img width="681" height="175" alt="image" src="https://github.com/user-attachments/assets/953ddde2-be6f-4e2f-be31-f24023b0551a" />

Reply:
Very wired, when I check out `chore: create test for sfn generation (4/6 do fail)` I get that:

<img width="705" height="694" alt="Screenshot 2025-09-22 at 08 24 23" src="https://github.com/user-attachments/assets/47f47a62-3848-4684-87cf-79282920e20b" />